### PR TITLE
Fix `_cumsum` helper function in multi-gpu

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes https://github.com/Lightning-AI/torchmetrics/issues/2608

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2636.org.readthedocs.build/en/2636/

### CODE REVIEW

File: src/torchmetrics/utilities/data.py [hunk 1]
```diff
@@ -215,7 +215,8 @@ def _cumsum(x: Tensor, dim: Optional[int] = 0, dtype: Optional[torch.dtype] = No
             "Expect some slowdowns.",
             TorchMetricsUserWarning,
         )
-        return x.cpu().cumsum(dim=dim, dtype=dtype).cuda()
+        device = x.device
+        return x.cpu().cumsum(dim=dim, dtype=dtype).to(device)
```
Comment (Borda) [inline src/torchmetrics/utilities/data.py:219]:
```suggestion
        return x.cpu().cumsum(dim=dim, dtype=dtype).to(x.device)
```

File: src/torchmetrics/utilities/data.py [hunk 2]
```diff
@@ -215,7 +215,8 @@ def _cumsum(x: Tensor, dim: Optional[int] = 0, dtype: Optional[torch.dtype] = No
             "Expect some slowdowns.",
             TorchMetricsUserWarning,
         )
-        return x.cpu().cumsum(dim=dim, dtype=dtype).cuda()
+        device = x.device
```
Comment (Borda) [inline src/torchmetrics/utilities/data.py:218]:
```suggestion
```

**Review by Borda — APPROVED** [review]
Can we have a test for it please?

**Comment by SkafteNicki** [comment]
> Can we have a test for it please?

Added, but only because you asked nicely :P

**Review by baskrahmer — COMMENTED** [review]
LGTM!

**Pull request merged**